core/src/main/java/hudson/util/FormApply.java

```diff
@@ -87,4 +93,20 @@ public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object nod
             }
         };
     }
+
+    /**
+     * Generates the response for the asynchronous background form submission (AKA the Apply button),
+     * that will show a notification of certain type and with provided message. See <a href="https://github.com/jenkinsci/jenkins/blob/74610e024a6b8fd8feccdc51b8f7741aa6c30e3b/war/src/main/js/components/notifications/index.js#L13-L25">index.js</a> for supported notification types.
+     */
+    public static HttpResponseException showNotification(final String message, final String notificationType) {
```

Comment:
I wonder whether the second arg might be better off as an `enum` defined in this class. Then there's no need to `#htmlAttributeEscape` for lack of ability to put something else there.

Reply:
Thought about that too, will add.